### PR TITLE
Enhancement #65: overhaul weather generator

### PR DESCRIPTION
**Pull request by dschlaep**

### CODE REVIEW

**Comment by kpalmqui** [comment]
@dschlaep perhaps just a few minor comments to add above. Otherwise, I didn't see any problems!

File: R/swWeatherGenerator.R [hunk 1]
```diff
@@ -0,0 +1,757 @@
...
+      n_Dy <- sum(!x[, "WET_yesterday"], na.rm = na.rm)
+
+      c(
+        # `wetprob` calculated as the number of years with doy being wet
+        # given previous day is wet divided by the number of years with
+        # the previous day being wet
+        p_W_W = if (isTRUE(n_Wy > 0)) {
+            sum(x[, "WW"], na.rm = na.rm) / n_Wy
+          } else {
+            p_W
```
[the remark this answers is not in the document]
Comment (dschlaep) [reply R/swWeatherGenerator.R]: Addressed by 7d10a42

**Comment by dschlaep** [comment]
@kpalmqui Thanks for your comments. With commit 7d10a42, I added explanations as code comments to address your requests. Additionally, I pushed commit 3181f08 that corrects same cases of handling/propagation of missing input values and NAs in intermediate calculations. Please review again/approve. Thanks!

**Comment by kpalmqui** [comment]
@dschlaep looks great! Big thanks again.

**Pull request merged**